### PR TITLE
Farcasterd retrieve key from wallet before launching peerd

### PR DESCRIPTION
**Pull request by zkao**
Added  general mechanism for doing callbacks, such as: 
```
MakeOffer (orig req) -> return GetKeys (req) -> 
                                                         (rep) Keys -> continue MakeOffer (orig req)
```
Mechanism uses early return to exit original request, into the fallback path
requesting the keys, uses a RequestId to identify what pending_request to call
back, using recursion to continue the pending_requests following up receiving Keys back from wallet, such that the return is from the original request -- just proceding with execution

### CODE REVIEW

File: src/farcasterd/runtime.rs [hunk 1]
```diff
@@ -513,39 +527,53 @@ impl Runtime {
...
                     ));
-                } else {
-                    info!("Already listening on {}", &remote_addr);
-                    let msg = format!("Already listening on {}", &remote_addr);
-                    report_to.push((Some(source.clone()), Request::Progress(msg)));
-                }
+
                 let peer = internet2::RemoteNodeAddr {
-                    node_id: self.node_id()?,
+                    node_id: self.node_ids()[0], // FIXME shouldnt be 0 here
```
Comment (h4sh3d) [inline src/farcasterd/runtime.rs:576]: Does that FIXME should be handle now?

**Review by h4sh3d — APPROVED** [review]
LGTM

**Comment by zkao** [comment]
its good for now, problematic when node has multiple ids

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

File: src/farcasterd/runtime.rs (hunk 1 above)
Comment (zkao) [reply src/farcasterd/runtime.rs]: #922ce4d reports an error if there are multiple node_ids stating multiple ids are not supported

**Review by TheCharlatan — APPROVED** [review]

**Pull request merged**